python/tests/jet/test_gate.py

```diff
@@ -173,6 +173,43 @@ def test_data(self, gate, state, want_tensor):
         assert have_tensor.indices == want_tensor.indices
 
 
+class TestScale:
+    @pytest.mark.parametrize("gate", [jet.Hadamard(), jet.U2(1, 2)])
+    def test_attributes(self, gate):
+        """Tests that Scale returns the same attributes as the decorated gate."""
+        scaled_gate = jet.Scale(gate, scalar=1)
+        assert scaled_gate.name == gate.name
+        assert scaled_gate.num_wires == gate.num_wires
+        assert scaled_gate.params == gate.params
+
+    @pytest.mark.parametrize(
+        ["gate", "state", "scalar", "want_tensor"],
+        [
+            pytest.param(
+                jet.PauliX(),
+                jet.Tensor(indices=["1"], shape=[2], data=[1, 0]),
+                2,
+                jet.Tensor(indices=["0"], shape=[2], data=[0, 2]),
```

Comment:
An astute observation! The reason why the indices appear "swapped" becomes more clear when you look at the graphical representation of the tensor network for this circuit:
```
|0> ---(1)--- X ---(0)---
```
Contracting this tensor network yields a 1D tensor with an index 0:
```
(|0>, X) ---(0)---
```
Note that the indices of X default to `[0, 1]` which means that the "row" dimension of X is assigned to index 0 and the "column" dimension is assigned to index 1.  Since tensor contractions are (can be) defined in terms of the dot product, this means that the "input" to X has a 1 index and the "output" of X has a 0 index.

Does that make sense?

Reply:
Alright, that makes sense. Thanks! I'm still trying to wrap my head around how these tensor networks work. 😄